### PR TITLE
Add documentation for the application's API

### PR DESCRIPTION
**Pull request by cprokopiak**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    data: Bid[]
+}
+```
+
+However, there are some cases where the API will return a 400 status code. The
+error cases are:
+
+* The `aid` property of the request payload is undefined, null, or an empty string.
+* The `data` property of the request payload is undefined or null.
+* An error is returned when trying to put records onto the Kinesis Firehose stream in AWS.
```
Comment (cshilts) [inline docs/api.md]: Is this proper functionality or an error condition that will need to be tested or a note of the current state.

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Analytics Collector API
+
+The analytics collector application provides a simple API for receiving bid analytics from web pages.
+
+## API
+
+The following API endpoints are provided by the application:
+
+* POST `/analytics`
```
Comment (cshilts) [inline docs/api.md]: Is there any authentication that is required to post/get?

Comment (cprokopiak) [reply docs/api.md]: Right now, no, there is not authentication for the ingest APIs.

File: docs/api.md (hunk 1 above)
Comment (cprokopiak) [reply docs/api.md]: No, I don't think you will be able to test this functionality easily. This is the proper functionality, too. If there is something wrong with the Kinesis Firehose, the Firehose API method we call will fail, and that should propagate the error message back through our API.

The only way I can think of to test this right now is to remove/destroy the Kinesis Firehose. That might be possible in a Chaos Monkey type scenario, but we are nowhere close to being able to do that type of testing.

**Pull request merged**